cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go

```diff
@@ -46,17 +41,7 @@ func main() {
 	flag.Parse()
 	glog.Info("Kubernetes Elasticsearch logging discovery")
 
-	settings, err := clientcmd.LoadFromFile(*kubeconfig)
-	if err != nil {
-		glog.Fatalf("Error loading configuration from %s: %v", *kubeconfig, err.Error())
-	}
-
-	config, err := clientcmd.NewDefaultClientConfig(*settings, &clientcmd.ConfigOverrides{}).ClientConfig()
-	if err != nil {
-		glog.Fatalf("Failed to construct config: %v", err)
-	}
-
-	c, err := client.New(config)
+	c, err := client.NewInCluster()
```

Comment:
Fine for this PR, but I don't think we want to encourage a pattern where a program has to be in a cluster to run.  Should be some way to fall back to kubeconfig from a file, such as when doing ad-hoc testing.